apps/api/src/modules/job/job.service.ts

```diff
@@ -1,15 +1,78 @@
 import { Injectable } from '@nestjs/common'
 
-import { EditJobDto } from './job.dto'
+import { EditJobDto, GetJobCardWithMaxPageDto, SearchJobDto } from './job.dto'
+import { JobRepository } from './job.repository'
 
 @Injectable()
 export class JobService {
+  constructor(private repository: JobRepository) {}
+
   create(createJobDto: EditJobDto) {
     return 'This action adds a new job'
   }
 
-  findAll() {
-    return `This action returns all job`
+  convertRequestToParams(searchJobDto: SearchJobDto) {
+    const params = {}
+    if (searchJobDto.limit) {
+      params['take'] = Number(searchJobDto.limit)
+    }
+    if (searchJobDto.page) {
+      params['skip'] = (searchJobDto.page - 1) * searchJobDto.limit
+    }
+    return params
+  }
+  async findAll(searchJobDto: SearchJobDto) {
+    //check searchJobDTO received
+    if (
+      searchJobDto.limit &&
+      searchJobDto.page &&
+      Object.keys(searchJobDto).length === 2
+    ) {
```

Comment:
I think we don't need to use this if. 
We might be able to use undefined with [this docs](https://www.prisma.io/docs/concepts/components/prisma-client/null-and-undefined)